iib/workers/tasks/legacy.py

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# This file can be deleted once OMPS is retired
+import logging
+import os
+import shutil
+import tempfile
+
+import requests
+from operatorcourier import api, errors
+
+from iib.exceptions import IIBError
+from iib.workers.api_utils import get_request, set_request_state, update_request
+from iib.workers.config import get_worker_config
+from iib.workers.tasks.utils import run_cmd, skopeo_inspect
+
+log = logging.getLogger(__name__)
+
+
+def get_legacy_support_packages(bundles):
+    """
+    Checks if legacy support is required for any packages in the bundles.
+
+    :param list<str> bundles: a list of strings representing the pull specifications of the bundles
+        to add to the index image being built.
+    :return: a set of packages that require legacy support
+    :rtype: set
+    """
+    packages = set()
+    for bundle in bundles:
+        skopeo_out = skopeo_inspect(f'docker://{bundle}')
+        if skopeo_out['Labels'].get('com.redhat.delivery.backport', False):
+            packages.add(skopeo_out['Labels']['operators.operatorframework.io.bundle.package.v1'])
+
+    return packages
+
+
+def _podman_login():
+    """
+    Login into the registry to be able to access private repos.
+
+    :raises iib.exceptions.IIBError: if the login fails
+    """
+    conf = get_worker_config()
+    log.info('Logging in %s using podman', conf['iib_registry'])
+    username, password = conf['iib_registry_credentials'].split(':')
+    run_cmd(
+        ['podman', 'login', '-u', username, '-p', password, conf['iib_registry']],
+        exc_msg=f'Failed to login into the registry {conf["iib_registry"]}',
+    )
+
+
+def _verify_package_info(package_dir, from_index):
+    """
+    Verifies if the exported package info is generated correctly
+
+    :param str package_dir: path to the exported package directory
+    :param str from_index: the pull specification of the image image
+    :raises iib.exceptions.IIBError: if the generated package info is incorrect or missing
+    """
+    package_name = package_dir.split('/download-', 1)[-1]
+    log.info('Verifying package_name %s', package_name)
+    # opm does not fail when the package is missing in the index image, hence we
+    # check the number of generated files. If it's equal to 1, that means only an empty
+    # `package.yaml` file is generated and the package is missing
+    if len(os.listdir(package_dir)) == 1:
+        raise IIBError(f'package {package_name} is missing in index image {from_index}')
+
+    try:
+        output = api.build_and_verify(source_dir=package_dir)
```

Comment:
OMPS will run the same verification command. Could we simply rely on it?

Reply:
I haven't looked at the code of OMPS, to find out if it runs this immediately when the zipfile is submitted. So I added it here just to be sure. But I can check if OMPS does this check before returning the response and remove it from here if you want.

Reply:
ok, looks like we can rely on OMPS to check this. I'll remove it.

Reply:
https://github.com/release-engineering/operators-manifests-push-service/blob/master/omps/quay.py#L312

Reply:
As we discussed, having this extra layer of verification doesn't do harm. So let's keep it. :)

Reply:
As discussed, let's remove this since OMPS does this for us and we can avoid adding more dependencies.